nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -595,7 +595,8 @@ in
         mkOption {
           type = types.package;
           default = cfg.host.pkgs.qemu_kvm;
-          example = "pkgs.qemu_test";
+          defaultText = literalExpression "config.virtualisation.host.pkgs.qemu_vm";
```

Comment:
I think you meant `"config.${options.virtualisation.host.pkgs}.qemu_vm"`, but that fails with `attribute 'host' missing` for some reason.

Reply:
Yeah that doesn't work either because `virtualisation` is not an option.

The other error comes from the fact that we [override](https://github.com/NixOS/nixpkgs/blob/6a36558f5227f728d4fa433ab8ec1b80bebe4f50/nixos/modules/misc/documentation.nix#L43-L45) the `options` argument with the outer `options` so that eager modules have access to all options, but indeed "all" doesn't include the `qemu-vm.nix` module by default.

I looked into doing something like `options = lib.recursiveUpdate allOpts eagerOpts`, where `eagerOpts = scrubbedEval.options`, but this runs into infinite recursion because those are the *`checked`* options. We would need to be able to access the unchecked `options` fixed point, which could be done either by adding `uncheckedOptions = options;` [here](https://github.com/NixOS/nixpkgs/blob/6a36558f5227f728d4fa433ab8ec1b80bebe4f50/lib/modules.nix#L328-L333) or by allowing `specialArgs` to be a function that accepts `options`.

In any case this is way beyond the scope of this PR.